### PR TITLE
Add module to create images from Rackspace servers

### PR DESCRIPTION
**Pull request by vladymyrs**
Moved this here as were advised in https://github.com/ansible/ansible-modules-core/pull/1483

### CODE REVIEW

**Comment by sivel** [comment]
I'll look into this as soon as possible.  I will write some integration tests so that I can fully test the module.  Once that is complete I should have a list of any needed updates and will work towards getting this merged.

**Comment by vladymyrs** [comment]
Cool, thanks.

**Comment by robynbergeron** [comment]
Hey @sivel -- do we really need integration tests here just to test that it works, so it can go into extras? Can we just test that it works? :) I hate to see this sit here waiting on that.

**Comment by sivel** [comment]
My personal opinion is that if I am expected to support these, which I am, which extends just beyond ansible, but to my employer, than yes, integration tests are required.

**Comment by vladymyrs** [comment]
Should I write some? And where should I push the tests don't see section in documentation about contributing tests.

Thanks

**Comment by robynbergeron** [comment]
@sivel: Thanks for the context. Much appreciated :)

@vladymyrs -- I'm not sure offhand how @sivel wants to approach this, but there is some information here about integration tests here:

http://docs.ansible.com/ansible/developing_test_pr.html#for-those-about-to-test-we-salute-you

and the associated code is here -- https://github.com/ansible/ansible/tree/devel/test/integration -- I'm sure with a bit of poking around in here you can get a better idea of what to look at. I myself am not entirely up to speed yet on all the details here :) but if @sivel has any further guidance I'm sure he will provide it. 

Thanks to both of you for continuing to keep this one going :)

File: cloud/rackspace/rax_images.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+# a documentation fragment located in ansible.utils.module_docs_fragments
+DOCUMENTATION = '''
+---
+module: rax_images
+short_description: Manipulate Rackspace Cloud Server Images
+description:
+  - Manipulate Rackspace Cloud Server Images
+version_added: "2.0"
+options:
+  name:
```
Comment (sivel) [inline cloud/rackspace/rax_images.py:27]: Please collapse `id` and `name` into a single `server` such as what is done in `rax_cbs_attachments`:  https://github.com/ansible/ansible-modules-core/blob/devel/cloud/rackspace/rax_cbs_attachments.py#L37-L41

File: cloud/rackspace/rax_images.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+            servers = cs.servers.list(search_opts=search_opts)
+        except Exception, e:
+            module.fail_json(msg='%s' % e.message)
+
+    elif server_id:
+        servers = []
+        try:
+            servers.append(cs.servers.get(server_id))
+        except Exception, e:
+            pass
```
Comment (sivel) [inline cloud/rackspace/rax_images.py:98]: With collapsing `name` and `id` into server, use `rax_find_server` like it is used at https://github.com/ansible/ansible-modules-core/blob/devel/cloud/rackspace/rax_cbs_attachments.py#L112

File: cloud/rackspace/rax_images.py [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+    if len(servers) > 1:
+        module.fail_json(msg='Multiple servers found matching provided '
+                             'search parameters')
+    elif not servers:
+        module.fail_json(msg='Failed to find a server matching provided '
+                             'search parameters')
```
Comment (sivel) [inline cloud/rackspace/rax_images.py:105]: These if statements are not needed when using `rax_find_server`

**Comment by sivel** [comment]
This module should support a state of `absent` and `present`.  Additionally, both of these states should be idempotent.  Currently the only state it supports is creating an image, and it always does it, and is not idempotent (meaning if already created, it should not create the image again, unless perhaps forced to do so)

**Comment by gregdek** [comment]
@vladymyrs After some review, it looks as though this PR needs some revisions. 

Please make the suggested revisions, and when you're done, please add a comment with the text "ready_for_review" and we will continue with the review process.  Thanks!

**Comment by vladymyrs** [comment]
Sure will work on fixes your comments here. Will take some time, since I'm not using this module anymore.

**Pull request closed without merging**